x-pack/solutions/observability/plugins/observability_agent/server/tools/log_rate_analysis/log_rate_analysis.ts

```diff
@@ -0,0 +1,130 @@
+import type { WindowParameters } from '@kbn/aiops-log-rate-analysis/window_parameters';
+import type {
+  ObservabilityAgentPluginStart,
+  ObservabilityAgentPluginStartDependencies,
+} from '../../types';
+
+export const OBSERVABILITY_RUN_LOG_RATE_ANALYSIS_TOOL_ID = 'observability.run_log_rate_analysis';
+
+const dateRangeSchema = z.object({
+  from: z
+    .string()
+    .describe('Start of the time window expressed with Elastic date math. Example: `now-15m`'),
+  to: z
+    .string()
+    .describe('End of the time window expressed with Elastic date math. Example: `now`.'),
+});
+
+const logRateAnalysisSchema = z.object({
+  index: z
+    .string()
+    .describe(
+      'Concrete index, data stream, or alias to analyze (for example `logs-payments.api-default`).'
+    ),
+  timeFieldName: z
+    .string()
+    .describe(
+      'Timestamp field used to build the baseline/deviation windows. Defaults to `@timestamp`.'
+    )
+    .optional(),
+  baseline: dateRangeSchema.describe(
+    'Time range representing “normal” behavior that the deviation window will be compared against.'
+  ),
+  deviation: dateRangeSchema.describe(
+    'Time range representing the time period with unusual behavior.'
+  ),
+});
+
+export function createRunLogRateAnalysisTool({
+  core: _core,
+  logger,
+}: {
+  core: CoreSetup<ObservabilityAgentPluginStartDependencies, ObservabilityAgentPluginStart>;
+  logger: Logger;
+}): StaticToolRegistration<typeof logRateAnalysisSchema> {
+  const toolDefinition: BuiltinToolDefinition<typeof logRateAnalysisSchema> = {
+    id: OBSERVABILITY_RUN_LOG_RATE_ANALYSIS_TOOL_ID,
+    type: ToolType.builtin,
+    description:
+      "Explain log spikes or dips by comparing two time windows (baseline vs deviation) and letting Kibana's log rate analysis surface the most significant field/value cohorts and categorized messages. Use this whenever you need to answer “what changed in the logs?” for a given index.",
+    schema: logRateAnalysisSchema,
+    tags: ['observability', 'logs'],
+    handler: async ({ index, timeFieldName = '@timestamp', baseline, deviation }, context) => {
+      try {
+        const esClient = context.esClient.asCurrentUser;
+
+        const windowParameters: WindowParameters = {
+          baselineMin: parseDate(baseline.from),
+          baselineMax: parseDate(baseline.to, { roundUp: true }),
+          deviationMin: parseDate(deviation.from),
+          deviationMax: parseDate(deviation.to, { roundUp: true }),
+        };
+
+        const deviationDurationMs = windowParameters.deviationMax - windowParameters.deviationMin;
+        const intervalMinutes = Math.max(1, Math.round(deviationDurationMs / 60000));
+
+        const response = await fetchLogRateAnalysisForAlert({
+          esClient,
+          arguments: {
+            index,
+            alertStartedAt: new Date(windowParameters.deviationMax).toISOString(),
```

Comment:
Should this be `windowParameters.deviationMin` so it's the start of the deviation?

Posting here this annotated screenshot from back when we did log rate analysis for contextual insights:

<img width="3378" height="620" alt="Image" src="https://github.com/user-attachments/assets/4afe4bdb-5b7d-4b07-994d-a81abb75688e" />

https://github.com/elastic/kibana/pull/187690#discussion_r1732677385